src/storage/kv/transaction.rs

```diff
@@ -1493,4 +1488,107 @@ mod tests {
         let results = txn.scan(range, None).unwrap();
         assert_eq!(results.len(), 0);
     }
+
+    #[tokio::test]
+    async fn transaction_delete_and_scan_test() {
+        let (store, _) = create_store(false);
+
+        // Define key-value pairs for the test
+        let key1 = Bytes::from("k1");
+        let value1 = Bytes::from("baz");
+
+        {
+            // Start a new read-write transaction (txn1)
+            let mut txn1 = store.begin().unwrap();
+            txn1.set(&key1, &value1).unwrap();
+            txn1.commit().await.unwrap();
+        }
+
+        {
+            // Start a read-only transaction (txn)
+            let mut txn = store.begin().unwrap();
+            txn.get(&key1).unwrap();
+            txn.delete(&key1).unwrap();
```

Comment:
Shouldn't this operation fail under R-only?

Reply:
This is a transaction that commits, do you mean this has to be Read only?